dev/src/Command/AddComponentCommand.php

```diff
@@ -201,4 +248,160 @@ private function loadProtoContent(string $proto): string
+            '-v',
+            $googleApisGenDir . ':/googleapis-gen',
+            '-w',
+            '/repo',
+            '--env',
+            'HOME=/tmp',
+            self::OWLBOT_CLI_IMAGE,
+            'copy-code',
+            sprintf('--config-file=%s/.OwlBot.yaml', $componentName),
+            '--source-repo=/googleapis-gen'
+        ];
+        return $this->runCommand($command);
+    }
+
+    private function owlbotCopyBazelBin(string $componentName, string $googleApisGenDir): string
+    {
+        list($userId, $groupId) = $this->getUserAndGroupId();
+        $command = [
+            'docker',
+            'run',
+            '--rm',
+            '--user',
+            sprintf('%s::%s', $userId, $groupId),
+            '-v',
+            $this->rootPath . ':/repo',
+            '-v',
+            $googleApisGenDir . '/bazel-bin:/bazel-bin',
+            self::OWLBOT_CLI_IMAGE,
+            'copy-bazel-bin',
+            sprintf('--config-file=%s/.OwlBot.yaml', $componentName),
+            '--source-dir',
+            '/bazel-bin',
+            '--dest',
+            '/repo'
+        ];
+        return $this->runCommand($command);
+    }
+
+    private function postProcess()
+    {
+        list($userId, $groupId) = $this->getUserAndGroupId();
+        $command = [
+            'docker',
+            'run',
+            '--rm',
+            '--user',
+            sprintf('%s::%s', $userId, $groupId),
+            '-v',
+            $this->rootPath . ':/repo',
+            '-w',
+            '/repo',
+            self::OWLBOT_PHP_IMAGE
+        ];
+        return $this->runCommand($command, null, null, true);
+    }
+
+    private function isProductHomePageMissing(?string $productHomePage): bool
+    {
+        $client = new Client();
+        if (!$productHomePage) {
+            return true;
+        }
+
+        try {
+            $response = $client->request('GET', $productHomePage, ['http_errors' => false]);
+            return $response->getStatusCode() === 404;
+        } catch (Exception $e) {
+            // Handle error gracefully.
+            return true;
+        }
+    }
+
+    private function runCommand(
+        array $command,
+        ?string $workDir = null,
+        ?string $input = null,
+        bool $errorOutput = false
+    ): string
+    {
+        $process = new Process($command);
+        if (!is_null($workDir)) {
+            $process->setWorkingDirectory($workDir);
+        }
+        if (!is_null($input)) {
+            $process->setInput($input);
+        }
+        // `mustRun` will throw a ProcessFailedException if the process
+        // couldn't be executed successfully.
+        $process->mustRun();
+        if ($errorOutput) {
+            return $process->getErrorOutput();
+        }
+        return $process->getOutput();
+    }
+
+    private function getHomePageFromDocsUrl(?string $url) {
+        $pattern = "/^(https?:\/\/)?([^\/]+)\/(.*?)\/docs\/(.+)/";
```

Comment:
I agree that regex is a bit overkill here - I'd either stick with your use of `explode`, or use `str` functions.

Reply:
No problem, thank you for the suggestion!

Reply:
Changed it to explode as its a concise way.